dap_job_quality/pipeline/prodigy/sentence_classifier_recipe.py

```diff
@@ -133,13 +158,36 @@ def make_tasks(
     "job_ad_sent_cat",
     dataset=("The dataset to use", "positional", None, str),
     source=("The source data as a .jsonl file", "positional", None, Path),
+    unsegmented=("Don't split sentences", "flag", "U", bool),
 )
-def custom_ner(dataset, source):
+def custom_ner(dataset, source, unsegmented: bool = False):
     # Initialize the Prodigy stream
     blocks = [{"view_id": "ner_manual"}, {"view_id": "text_input"}]
 
     stream = JSONL(source)
 
+    if not unsegmented:
+        # Custom way to split into chunks of a certain size
+        # its not ideal if these are too big (the model struggles)
+        # or too small (it's hard to label)
+        def split_text(adverts, chunk_size=5):
+            for advert in adverts:
+                text = advert["text"]
+                id = advert["meta"]
+                sentences = text.split(".")
```

Comment:
Avoid splitting up e.g "£14.70 per hour":
```suggestion
                sentences = re.split(r'(?<!\d)\.(?!\d)', text)
```